src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java

```diff
@@ -112,80 +107,66 @@ public Bucket update(Bucket bucket) {
                 statement.executeUpdate();
             }
         } catch (SQLException e) {
-            throw new RuntimeException(e);
+            throw new DataProcessingException("Failed to update bucket: " + e);
         }
         return bucket;
     }
 
-    private void deleteOldBucket(Long bucketId) {
+    private void deleteOldBucket(Long bucketId) throws DataProcessingException {
         long userId = bucketId;
         String query = "DELETE FROM internet_shop.buckets WHERE user_id=?;";
         try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
             preparedStatement.setLong(1, userId);
             preparedStatement.executeUpdate();
         } catch (SQLException e) {
-            throw new RuntimeException();
+            throw new DataProcessingException("Failed to delete old bucket: " + e);
         }
     }
 
     @Override
-    public boolean delete(Bucket bucket) {
+    public boolean delete(Bucket bucket) throws DataProcessingException {
         long userId = bucket.getUserId();
         String query = "DELETE FROM internet_shop.buckets WHERE user_id=?;";
         try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
             preparedStatement.setLong(1, userId);
             int rs = preparedStatement.executeUpdate();
```

Comment:
execute() повертає true/false
можливо краще використовувати його?

Reply:
No. Because the execute method returns a boolean type value where true indicates that the ResultSet s returned which can later be extracted and false indicates that the integer or void value is returned. After delete I dont't  have resultSet